systempoweraction/src/main/java/com/stream_pi/systempoweraction/SystemPowerAction.java

```diff
@@ -0,0 +1,163 @@
+import com.stream_pi.util.platform.Platform;
+import com.stream_pi.util.exception.MinorException;
+import com.stream_pi.util.version.Version;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+public class SystemPowerAction extends NormalAction
+{
+    public SystemPowerAction()
+    {
+        setName("Power Action");
+        setAuthor("quimodotcom");
+        setHelpLink("https://github.com/quimodotcom/essential-actions");
+        setVersion(new Version(1,0,0));
+        setCategory("System");
+        
+        states = new ArrayList<>();
+        states.addAll(Arrays.asList(
+                new ListValue("Shutdown"),
+                new ListValue("Restart"),
+                new ListValue("Sleep")
+        ));
+    }
+    
+    private final ArrayList<ListValue> states;
+    
+    @Override
+    public void initProperties() throws MinorException
+    {
+        Property status = new Property("state", Type.LIST);
+        status.setListValue(states);
+        status.setDisplayName("Power State");
+
+        addClientProperties(status);
+    }
+    
+    @Override
+    public void onActionClicked() throws MinorException
+    {
+        Platform plat = getPlatform();
+        String state = states.get(getClientProperties().getSingleProperty("state").getSelectedIndex()).getName().toString();
+        switch (state)
+        {
+            case "Shutdown":
+                Shutdown(plat);
+                break;
+            
+            case "Restart":
+                Restart(plat);
+                break;
+                
+            case "Sleep":
+                Sleep(plat);
+                break;     
+        }
+    }
+    
+    public void Shutdown(Platform plat) throws MinorException
+    {
+        String shutdownCommand = new String();
+        try{
+            if (null != plat) switch (plat) {
+            case MAC:
+                Process p = Runtime.getRuntime().exec
+                        ("/bin/bash");
+                    String command = "osascript -e 'tell application \"System Events\"' "
+                            + " -e \"shut down\" -e 'end tell'";
+                    OutputStream stdin = p.getOutputStream();
+                    stdin.write( command.getBytes() );
+                    stdin.flush();
+                    stdin.close();
+                break;
+            case WINDOWS:
+                shutdownCommand = "shutdown.exe -s -t 0";
+                break;
+            case LINUX:
+                shutdownCommand = "shutdown -h now";
+            case UNKNOWN:
+                throw new RuntimeException("Unsupported operating system.");
+            default:
+                break;
+                }
+            Runtime.getRuntime().exec(shutdownCommand);
+            } catch (IOException e){
+                    throw new MinorException(e.getMessage());
+            }
+    }
+    
+    public void Sleep(Platform plat) throws MinorException
```

Comment:
Not following java method name convention - method name should start with a lower case character

Reply:
done